lib/db/set.go

```diff
@@ -168,6 +168,10 @@ type Snapshot struct {
 func (s *FileSet) Snapshot() (*Snapshot, error) {
 	opStr := fmt.Sprintf("%s Snapshot()", s.folder)
 	l.Debugf(opStr)
+
+	s.updateMutex.Lock()
+	defer s.updateMutex.Unlock()
```

Comment:
Given these snapshots can happen quite a lot, during times of heavy writes (e.g. new remote) there might be some lock contention. Thus maybe worth making it an RW-mutex, such that all snapshots can happen at once with a read-lock as soon as a (batch) write with a write-lock is done.